pandas/tests/indexes/multi/test_get_set.py

```diff
@@ -0,0 +1,649 @@
+    exp = CategoricalIndex([1, 2, 3, 1, 2, 3])
+    tm.assert_index_equal(index.get_level_values(1), exp)
+
+
+def test_getitem(idx):
+    # scalar
+    assert idx[2] == ('bar', 'one')
+
+    # slice
+    result = idx[2:5]
+    expected = idx[[2, 3, 4]]
+    assert result.equals(expected)
+
+    # boolean
+    result = idx[[True, False, True, False, True, True]]
+    result2 = idx[np.array([True, False, True, False, True, True])]
+    expected = idx[[0, 2, 4, 5]]
+    assert result.equals(expected)
+    assert result2.equals(expected)
+
+
+def test_getitem_group_select(idx):
+    sorted_idx, _ = idx.sortlevel(0)
+    assert sorted_idx.get_loc('baz') == slice(3, 4)
+    assert sorted_idx.get_loc('foo') == slice(0, 2)
+
+
+def test_get_value_duplicates():
+    index = MultiIndex(levels=[['D', 'B', 'C'],
+                               [0, 26, 27, 37, 57, 67, 75, 82]],
+                       labels=[[0, 0, 0, 1, 2, 2, 2, 2, 2, 2],
+                               [1, 3, 4, 6, 0, 2, 2, 3, 5, 7]],
+                       names=['tag', 'day'])
+
+    assert index.get_loc('D') == slice(0, 3)
+    with pytest.raises(KeyError):
+        index._engine.get_value(np.array([]), 'D')
+
+
+def test_get_level_values_all_na():
+    # GH 17924 when level entirely consists of nan
+    arrays = [[np.nan, np.nan, np.nan], ['a', np.nan, 1]]
+    index = pd.MultiIndex.from_arrays(arrays)
+    result = index.get_level_values(0)
+    expected = pd.Index([np.nan, np.nan, np.nan], dtype=np.float64)
+    tm.assert_index_equal(result, expected)
+
+    result = index.get_level_values(1)
+    expected = pd.Index(['a', np.nan, 1], dtype=object)
+    tm.assert_index_equal(result, expected)
+
+
+def test_get_level_values_int_with_na():
+    # GH 17924
+    arrays = [['a', 'b', 'b'], [1, np.nan, 2]]
+    index = pd.MultiIndex.from_arrays(arrays)
+    result = index.get_level_values(1)
+    expected = Index([1, np.nan, 2])
+    tm.assert_index_equal(result, expected)
+
+    arrays = [['a', 'b', 'b'], [np.nan, np.nan, 2]]
+    index = pd.MultiIndex.from_arrays(arrays)
+    result = index.get_level_values(1)
+    expected = Index([np.nan, np.nan, 2])
+    tm.assert_index_equal(result, expected)
+
+
+def test_get_level_values_na():
+    arrays = [[np.nan, np.nan, np.nan], ['a', np.nan, 1]]
+    index = pd.MultiIndex.from_arrays(arrays)
+    result = index.get_level_values(0)
+    expected = pd.Index([np.nan, np.nan, np.nan])
+    tm.assert_index_equal(result, expected)
+
+    result = index.get_level_values(1)
+    expected = pd.Index(['a', np.nan, 1])
+    tm.assert_index_equal(result, expected)
+
+    arrays = [['a', 'b', 'b'], pd.DatetimeIndex([0, 1, pd.NaT])]
+    index = pd.MultiIndex.from_arrays(arrays)
+    result = index.get_level_values(1)
+    expected = pd.DatetimeIndex([0, 1, pd.NaT])
+    tm.assert_index_equal(result, expected)
+
+    arrays = [[], []]
+    index = pd.MultiIndex.from_arrays(arrays)
+    result = index.get_level_values(0)
```

Comment:
so out of this pls move to test_indexing:

get_indexer, getitem and like. this test should be for get/set levels, labels, names, get_level_values and so on

Reply:
I think this one is done, but please go over it one last time.

Reply:
I also moved `test_get_unique_index` to test_uniq_dups.

Reply:
i indicated some more to move